### PR TITLE
Added non printable characters rendering

### PR DESCRIPTION
**Pull request by RaresCon**
I added non printable characters rendering using the new key bind CTRL+P, that also lets you hide them. Now whitespaces (`·`), tabs (`├──┤`), linefeeds (`␊`) and other control characters will be rendered (I encountered a problem here, some terminals are not in fact able to render UTF-8 characters, so I tested the code of the character that appears on render and all codes are correct). The problem now is that nightly formatting is messing around with all if statements and other declarations that I tried to write on a single line, so the number of lines is above limit by almost 50 lines. I tried modifying the fmt config file, adding other rules that would help reduce the number of lines, but I couldn't force it to format everything more compressed. Even though I can compress them by hand, the code won't pass the tests. Any tips for refactoring the code would be awesome.

Issue: #2 

Non printable characters demo

### CODE REVIEW

File: src/row.rs [hunk 1]
```diff
@@ -176,13 +194,35 @@ impl Row {
...
+                        '\x0D' => rendered_char = '␍',
+                        // null
+                        '\x00' => rendered_char = '␀',
+                        // bell
+                        '\x07' => rendered_char = '␇',
+                        // backspace
+                        '\x08' => rendered_char = '␈',
+                        // escape
+                        '\x1B' => rendered_char = '␛',
+                        _ => rendered_char = '?',
```
Comment (alexandruradovici) [inline src/row.rs:223]: What about using a constant `HashMap`?

File: src/config.rs [hunk 2]
```diff
@@ -20,12 +20,20 @@ pub struct Config {
     pub message_dur: Duration,
     /// Whether to display line numbers.
     pub show_line_num: bool,
+    /// Whether to display nonprintable characters.
+    pub non_printable: bool,
```
Comment (alexandruradovici) [inline src/config.rs:24]: Use a shorter name here so that the default function fits into one line.

File: src/editor.rs [hunk 3]
```diff
@@ -21,6 +21,7 @@ const PASTE: u8 = ctrl_key(b'V');
 const DUPLICATE: u8 = ctrl_key(b'D');
 const EXECUTE: u8 = ctrl_key(b'E');
 const REMOVE_LINE: u8 = ctrl_key(b'R');
+const NON_PRINTABLE: u8 = ctrl_key(b'P');
```
Comment (alexandruradovici) [inline src/editor.rs:24]: Use tuple assignments here like:

```rust
const (EXECUTE, REMOVE_LINE, NON_PRINTABLE): (u8, u8, u8) = (ctrl_key(b'E'), ctrl_key(b'R'), ctrl_key(b'P'));
```

**Review by alexandruradovici — COMMENTED** [review]
Some tips to delete some lines.

File: src/row.rs (hunk 1 above)
Comment (RaresCon) [reply src/row.rs:223]: I tried to use it, but it doesn't compile as all methods for creating a HashMap are not `const` and other options like a sorted Map of characters would add too many lines.

File: src/config.rs (hunk 2 above)
Comment (RaresCon) [reply src/config.rs:24]: I think the formatting doesn't care about the length of the default function when there are too many arguments. I tried to shorten all names and it still moves them on a new line.

**Comment by RaresCon** [comment]
I've seen that the number of lines for `kibi` is now 1023 (too close). I decided that I will try to refactor as much as possible the code to reduce the number of lines before trying to implement this feature. Thank you again!

**Comment by alexandruradovici** [comment]
Waiting for the refactoring pull request 😄

**Pull request closed without merging**